src/parse.c

```diff
@@ -8,12 +8,14 @@
 
 int git_parse_ctx_init(git_parse_ctx *ctx, const char *content, size_t content_len)
 {
-	if (content_len)
+	if (content && content_len) {
 		ctx->content = content;
-	else
-		ctx->content = NULL;
+		ctx->content_len = content_len;
+	} else {
+		ctx->content = "";
+		ctx->content_len = 0;
+	}
```

Comment:
Yeah, this fixes the issue. I'm wondering though if we should even make it a bug to pass in `NULL` to `git_parse_ctx_init` by adding an assert. That'd require us to review all callers, though

Reply:
That was my initial idea. In a (non-published) earlier version, I added an assertion for catching calls to `git_parse_ctx_init` with `NULL` for `context`. However, I quickly ran into places where this fails.

For example, functions for parsing stuff built upon `git_parse_ctx` would also happily pass `NULL` for `context`. I even run into a test case explicitly checking for the ability to parse a patch from an empty _and_ a `NULL` buffer:

https://github.com/libgit2/libgit2/blob/3be7301151e44537d2423386dc71f1d3e496c664/tests/apply/fromfile.c#L448

Reply:
Huh, so be it then. Thanks for clarifying!